backend/app/api/endpoints/keys.py

```
@@ -0,0 +1,31 @@
+    except Exception as e:
+        logger.error(f"Error setting API keys: {e}")
+        raise HTTPException(status_code=500, detail=str(e))
+
+@router.get("", response_model=APIKeyStatus)
+async def get_key_status():
+    """Retrieve which API keys have been set"""
+    try:
+        return api_key_service.get_status()
+    except Exception as e:
+        logger.error(f"Error getting API key status: {e}")
+        raise HTTPException(status_code=500, detail=str(e))
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)